### ISSUE TITLE
Circuit Breaker suggestions

### ISSUE THREAD

**Issue by mattcan** [maumercado/codigo.sh#1]
I don't have any suggestions for content, I think it is really great! I do have a few grammar and readability suggestions though.

> Commonly, making remote calls it's a natural part of any system created, but those system can fail by any number of reasons. The circuit breaker design pattern can help in these specific types of failures.

It is hard to understand what types of failure are being referenced when you say "these specific types". I think the problem here is actually the order of the content:

```
<Narrow problem to solve>, <broad statement>. <The technique to solve the narrow problem>.
```

What might work better is:

```
<Broad statement>, <narrow down on problem to solve>. <The technique to solve the narrow problem>.
```

---

> Handling remote API failures

This section feels like an advertisement because it doesn't add any additional information about *this* post and restates the problem/solution from the opening. I think using this section as the opening would be better than the one you have currently.

If you do stick with this section, the commas are a bit off. I have reorganized them without changing any content:

> In the previous post, Failing successfully, it was shown that the best strategy when dealing with unknown errors was to fail properly and "noisily", instead ~than~of hiding them. This post will demonstrate how to properly handle failures when attempting to obtain information or execute procedures on a remote/external API.

---

> since it's software analog

I don't think analog should be there?

> its simply smarter

Should be *it's*. English is the worst.

---

> The variables for this state are, a failure count which as its name implies keeps count of how many times a service has failed to reply, a threshold of failures, which is simply a maximum amount of times an external service is allowed to fail, and a timeout value, which is the time in which a service should reply successfully.

This is too complex. When reading it I had to keep track of the previous phrase between commas so that I can jump back to recheck what definition was just read. I would suggest a bullet list instead.

---

I like the diagram after describing the Half-Open state. It may be beneficial to have simple open and closed diagrams for the respective states. I say this because the idea of "closed is normal, open is abnormal" is unnatural in english. I don't know if there is a better diagram than an electric circuit lol

---

In the 'Usage and customizations' section, one or two diagrams would ease the burden of understanding the scenario. [One to illustrate the previous three paragraphs](https://github.com/maumercado/codigo.sh/blame/master/src/routes/blog/posts/circuit-breakers.md#L97) and _maybe_ [one to illustrate how the circuit breaker solves it](https://github.com/maumercado/codigo.sh/blame/master/src/routes/blog/posts/circuit-breakers.md#L103). Related: [Aphantasia](https://en.wikipedia.org/wiki/Aphantasia)

I would also move the final paragraph in this section to the **Code example** section.

---

I am not a fan of 'here' links. Click [here](https://www.smashingmagazine.com/2012/06/links-should-never-say-click-here/#here-conceals-what-users-are-clicking) to see why.

I suggest rewording the **Code example** paragraph to highlight that you did a presentation on this but also go directly to the slide with the code. Some of us don't care about your achievements.

```markdown
I made a [presentation on the circuit breaker](https://cbr.maumercado.com) a while back and the presentation contains a [simple example of the circuit breaker implementation in NodeJS](https://cbr.maumercado.com/7).
```

---

The conclusion wraps things up really well, nicely done!

**Comment by maumercado** [comment]
Most comments should be dealt by #2

**Comment by mattcan** [comment]
Animations really did it, nice work!
